src/js/components/Marathon.jsx

```diff
@@ -224,6 +229,24 @@ var Marathon = React.createClass({
     );
   },
 
+  updateFilters: function (filters) {
+    var router = this.context.router;
+    var params = router.getCurrentParams();
+    var query = router.getCurrentQuery();
+
+    if (params != null && filters.filterText !== "") {
+      this.context.router.transitionTo("apps", params, query);
```

Comment:
There is the 'QueryParamMixin'.
Use the `setQueryParam`-function from the mixin.